onchain/src/social/account.cairo

```diff
@@ -48,6 +69,69 @@ pub mod SocialAccount {
         ) { // TODO: implement handle transfer logic
         }
     }
+
+    #[external(v0)]
+    impl SRC6Impl of interface::ISRC6<ContractState> {
+        fn __execute__(self: @ContractState, mut calls: Array<Call>) -> Array<Span<felt252>> {
+            let sender = get_caller_address();
+            assert(sender.is_zero(), 'Account: invalid caller');
+
+            // Check tx version
+            let tx_info = get_tx_info().unbox();
+            let version = tx_info.version;
+            if version != TRANSACTION_VERSION {
```

Comment:
What is the intention behind this code?

Reply:
to check the version of the code

Reply:
What QUERY_VERSION is supposed to mean?

Reply:
the QUERY_VERSION is restrict the account contract to only support the newest version of each type of transaction.

since 

   - Version 1 for invoke transactions
    - Version 1 for deploy_account transactions
    - Version 2 for declare transactions

Reply:
Would you please point me to the sources you are using? Documentation etc?

Reply:
In general you should rather use `openzeppelin::account::AccountComponent` (https://docs.openzeppelin.com/contracts-cairo/0.8.1/api/account).